nss-copper-dragonflies/BangazonAPI
# Customer Resource

# Description

I created the customer controller, class, and customer controller test for the customer resource. I also included query search parameters to include customer payment or customer product information for individual or multiple customers. The list of customers can be queried.

Fixes # 1

## Type of change


# Testing Instructions

1. `git fetch -a`
1. `git checkout brj-customer`
1. check that all tests for the customer controller are passing
1. run BangazonAPI
1. Open postman and perform a GET, POST, and PUT
1. past the following in the URL: `http://localhost:5000/api/customer?include=payment`. Ensure that the customers are returned with payment type details included
1. past the following in the URL: `http://localhost:5000/api/customer?include=product`. Ensure that the customers are returned with product details included
1. past the following in the URL: `http://localhost:5000/api/customer?q=al`. Ensure that customers with the corresponding query details are included. If following these instructions in order you should return information for the following customers:
```
[
  {
    "id": 1,
    "firstName": "Not Allison",
    "lastName": "Carter",
    "paymentTypes": [
      
    ],
    "products": [
      
    ]
  },
  {
    "id": 6,
    "firstName": "Dizzee",
    "lastName": "Rascal",
    "paymentTypes": [
      
    ],
    "products": [
      
    ]
  }
]
```



# Checklist:
